File: RomanToInteger/RomanToInteger.cpp
#include<string>
#include<unordered_map>
using namespace std;

class Solution {
public:
	int romanToInt(string s) {
		unordered_map<char, int> Roman;
		Roman['I'] = 1;
		Roman['V'] = 5;
		Roman['X'] = 10;
		Roman['L'] = 50;
		Roman['C'] = 100;
		Roman['D'] = 500;
		Roman['M'] = 1000;
		int n = s.size(), result = 0;
		for (int i = 0; i < n; i++) {
			if ((i < n - 1) && (Roman[s[i]] < Roman[s[i + 1]]))
				result -= Roman[s[i]];
			else
				result += Roman[s[i]];
		}
		return result;
	}
};

void main() {
	Solution A;
	printf("%d", A.romanToInt("DCXXI"));
}

File: RomanToInteger/RomanToInteger.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        resultmap = {'I':1, 'V':5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000}
        count = [resultmap[x] for x in s]
        result = sum(count)
        for i in range(len(count) - 1):
            if count[i] < count[i+1]:
                result -= 2*count[i]
        return result